File: src/main/java/de/cplaiz/activecraftcore/listener/SignListener.java
package de.cplaiz.activecraftcore.listener;

import de.cplaiz.activecraftcore.utils.ColorUtils;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.block.SignChangeEvent;

public class SignListener implements Listener {

    @EventHandler
    public void onSignChange(SignChangeEvent event) {
        for (int i = 0; i < 4; i++)
            event.setLine(i, ColorUtils.replaceColorAndFormat(event.getLine(i)));
    }

}

File: src/main/java/org/activecraft/activecraftcore/listener/SignListener.kt
package org.activecraft.activecraftcore.listener

import org.activecraft.activecraftcore.utils.replaceColorAndFormat
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.block.SignChangeEvent

class SignListener : Listener {
    @EventHandler
    fun onSignChange(event: SignChangeEvent) {
        for (i in 0..3) event.setLine(i, replaceColorAndFormat(event.getLine(i)!!))
    }
}